src/components/Layout/BannersUI.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div>
+    <k-banner
+      v-for="banner in formattedBanners"
+      :key="banner.id"
+      :class="bannerClasses"
+      :color="banner.color"
+      :icon="banner.icon"
+      :position="banner.position"
+      :dismissible="isDismissible(banner)"
+      @dismiss="isDismissible(banner) && $emit('dismiss', banner.id)"
+      :actions="banner.actions || []"
+    >
+      {{ $t(banner.message, banner.context) }}
+    </k-banner>
+  </div>
+</template>
+
+<script>
+import { Dialog } from 'quasar'
+import KBanner from '@/components/Layout/KBanner'
+
+export default {
+  components: { KBanner },
+  props: {
+    banners: {
+      type: Array,
+      required: true,
+    },
+  },
+  methods: {
+
+    isDismissible ({ id, dismissible }) {
+      return id && !dismissible
+    },
+
+    awaitingAgreement (agreement) {
+      return {
+        color: 'negative',
+        icon: 'pan_tool',
+        message: 'AGREEMENT.AGREE',
+        actions: [
+          {
+            label: this.$t('AGREEMENT.VIEW'),
+            handler: () => {
+              Dialog.create({
+                title: agreement.title,
+                message: agreement.content,
+                cancel: this.$t('BUTTON.CANCEL'),
+                ok: this.$t('BUTTON.AGREE'),
+              })
+                .then(() => this.$emit('agree', agreement.id))
+                .catch(() => {})
+            },
+          },
+        ],
+      }
+    },
+
+    playgroundGroupInfo () {
+      return {
+        color: 'positive',
+        icon: 'fa-child',
+        message: 'GROUP.PLAYGROUND_INFO',
+      }
+    },
+
+    bannerClasses () {
+      return this.$q.platform.is.mobile ? {
```

Comment:
Did you test the banner on mobile?

Reply:
No, I spent quite some time trying to set up cordova but there were loads of issues so I asked @nicksellen to fix them first. 

However, I didn't change the banner UI anyway, so with respect to the banners there's no difference to the current production version of karrot.

Reply:
I think the only banner that's ever shown on mobile is the group agreement one (the playground banner is hidden because it was blocking topbar access). So maybe we can ignore it.

You don't need Cordova, you can use the responsive design view to test it, e.g. https://developer.mozilla.org/en-US/docs/Tools/Responsive_Design_Mode